### DISCUSSION TITLE
Pathfinding algorithm (simulation)

### DISCUSSION THREAD

**Discussion by adamcate** [IEEE-Robotics-Club/MSU-NASA-Minds-2023#31] [Ideas]
1. Chunking
- Take the desired space (specified in meters height and length) and make hslices and vslices, measurement freq.
2. Generate initial search pattern w/ waypoints.
3. Obstacle detection with LIDAR
- Quadtree, etc, obstacle bounds (rectangular)
5. Recalculate path

**Comment by adamcate** [comment]
I think our first order of business is a good line rectangle intersection algorithm. Let me know who can help with this

**Reply by Fife** [reply]
Chunking algorithm and waypoint search pattern are in the main now. Going to start tackling LiDAR simulation
